requirements-dev.in

```diff
@@ -2,7 +2,7 @@
 bandit
 black
 coverage
-flake8
+flake8~=3.9
```

Comment:
Can you please add a comment about the reason for pinning to 3.9.x? (is it because of a conflict with `importlib-metadata` versions otherwise?)

Reply:
Sure, that change was in commit [0b7d1f6](https://github.com/jgreenbow-userful/aws-cdk-project-structure-python/commit/0b7d1f6c82d42698245806418c449ba8c6081bca). From that commit message:

> Pinned flake8 to prevent dependency conflict
>
> flake8>=4.0 pins importlib-metadata<4.3 as a dependency. This is incompatible
> with other packages requiring importlib-metadata>=4.12.

This is a python3.7 issue, as I believe flake8 for ^3.8 does not have the same pin. 

However, I will do a quick re-run on for pip-compile to see if this has been patched.

Reply:
I just ran another `pip-compile` on my Python 3.7.14 but same issue. Here is the output.

`> pip-compile --upgrade requirements-dev.in`
<details>
Could not find a version that matches importlib-metadata<4.3,==4.12.0,>=1.1.0 (from -c requirements.txt (line 41))
Tried: 0.0, 0.1, 0.1, 0.2, 0.2, 0.3, 0.3, 0.4, 0.4, 0.5, 0.5, 0.6, 0.6, 0.7, 0.7, 0.8, 0.8, 0.9, 0.9, 0.10, 0.10, 0.11, 0.11, 0.12, 0.12, 0.13, 0.13, 0.14, 0.14, 0.15, 0.15, 0.16, 0.16, 0.17, 0.17, 0.18, 0.18, 0.19, 0.19, 0.20, 0.20, 0.21, 0.21, 0.22, 0.22, 0.23, 0.23, 1.0.0, 1.0.0, 1.1.0, 1.1.0, 1.1.1, 1.1.1, 1.1.2, 1.1.2, 1.1.3, 1.1.3, 1.2.0, 1.2.0, 1.3.0, 1.3.0, 1.4.0, 1.4.0, 1.5.0, 1.5.0, 1.5.1, 1.5.1, 1.5.2, 1.5.2, 1.6.0, 1.6.0, 1.6.1, 1.6.1, 1.7.0, 1.7.0, 2.0.0, 2.0.0, 2.1.0, 2.1.0, 2.1.1, 2.1.1, 2.1.2, 2.1.2, 2.1.3, 2.1.3, 3.0.0, 3.0.0, 3.1.0, 3.1.0, 3.1.1, 3.1.1, 3.2.0, 3.2.0, 3.3.0, 3.3.0, 3.4.0, 3.4.0, 3.5.0, 3.5.0, 3.6.0, 3.6.0, 3.7.0, 3.7.0, 3.7.1, 3.7.1, 3.7.2, 3.7.2, 3.7.3, 3.7.3, 3.8.0, 3.8.0, 3.8.1, 3.8.1, 3.8.2, 3.8.2, 3.9.0, 3.9.0, 3.9.1, 3.9.1, 3.10.0, 3.10.0, 3.10.1, 3.10.1, 4.0.0, 4.0.0, 4.0.1, 4.0.1, 4.1.0, 4.1.0, 4.2.0, 4.2.0, 4.3.0, 4.3.0, 4.3.1, 4.3.1, 4.4.0, 4.4.0, 4.5.0, 4.5.0, 4.6.0, 4.6.0, 4.6.1, 4.6.1, 4.6.2, 4.6.2, 4.6.3, 4.6.3, 4.6.4, 4.6.4, 4.7.0, 4.7.0, 4.7.1, 4.7.1, 4.8.0, 4.8.0, 4.8.1, 4.8.1, 4.8.2, 4.8.2, 4.8.3, 4.8.3, 4.9.0, 4.9.0, 4.10.0, 4.10.0, 4.10.1, 4.10.1, 4.11.0, 4.11.0, 4.11.1, 4.11.1, 4.11.2, 4.11.2, 4.11.3, 4.11.3, 4.11.4, 4.11.4, 4.12.0, 4.12.0  

There are incompatible versions in the resolved dependencies:  
  importlib-metadata==4.12.0 (from -c requirements.txt (line 41))  
  importlib-metadata<4.3,>=1.1.0 (from flake8==5.0.4->-r requirements-dev.in (line 5))  
</details>

Would you prefer to pin `flake8` at a lower version?

Reply:
Thanks for checking! I thought that this is the issue, as I had it as well when bumping dependencies. I feel we should still support Python 3.7, for example, because [Lambda runtime](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html) still supports it. So let's keep the way you pinned flake8 in `requirements-dev.in`. Git blame will help to explain as you mentioned 😃.

![image](https://user-images.githubusercontent.com/4362270/190716453-83283370-6a17-48f8-9d64-e5ee30ec4a6d.png)